### PR TITLE
Implement render tag

### PR DESCRIPTION
**Pull request by bluzky**
**Options To render function**
  - `file_system`: a tuple of {FileSytemModule, options}. If this option is not specified, `Solid` use `Solid.BlankFileSystem` which raise error when you use `render` tag. You can use `Solid.LocalFileSystem` or implement your own file system which must implement method `read_template_file(template_path, options)` and return content of the template file.
  
  **Example**:

```elixir
fs = Solid.LocalFileSystem.new("/path/to/template/dir/")
Solid.render(template, vars, [file_system: {Solid.LocalFileSystem, fs}])
```

### CODE REVIEW

**Comment by bluzky** [comment]
@edgurgel Currently, in the render function for `render` tag, I return text `template not found` if no template found. What do you we should return here? Or we can add render options for this.

**Comment by edgurgel** [comment]
Thanks for starting this :tada:

I will review it properly this week but one thing I would add is that we should be aiming to have the same result that the Liquid gem would. That's why it's so important that we add as many `test/cases/` as possible as these tests are tested against Solid and against the Liquid gem. If they differ we know that our implementation is not doing the right thing.

So for example for the question regarding the template not being found I would say that we need to check what the Liquid gem does and probably do the exact same.

Same for `cwd` and `lookup_dir`. If people can't use these same options with the Liquid gem I don't think we should implement it like this. I would have a look on how they implement the resolver for files? Maybe we should just have a default implementation that mimics Liquid but opens up for people to define their own resolver (I have vague memories that Liquid does this as well).

Does this make sense?

**Comment by bluzky** [comment]
I read their liquid code and it pass `file_system` via register hash, I think we can pass file_system as an option to `render` function.

They implement some file system here.
https://github.com/Shopify/liquid/blob/50d1a2ffc9/lib/liquid/file_system.rb
And by default, it raise exception if no template found.

**Comment by bluzky** [comment]
@edgurgel I have re-implemented template look up for render tag by adding `LocalFileSystem` and pass it as an option to `render` function.

```elixir
fs = Solid.LocalFileSystem.new("/path/to/template/dir/")
Solid.render(template, vars, [file_system: {Solid.LocalFileSystem, fs}])
```
I also add test cases to cases directory.

File: lib/solid.ex [hunk 1]
```diff
@@ -51,6 +51,16 @@ defmodule Solid do
 
   @doc """
   It renders the compiled template using a `hash` with vars
+  To help `render` tag work correctly, you should specify `cwd` and `lookup_dir` options
```
Comment (edgurgel) [inline lib/solid.ex:54]: Is this still valid?

**Comment by edgurgel** [comment]
This is looking really good @bluzky ! I added just a comment on the docs.

I think we just need to upgrade the Liquid version we are installing here: 

https://github.com/edgurgel/solid/blob/e4a6c80d0d65476f0e790cc1a8dee29a5578cb70/.github/workflows/main.yml#L58

The latest version is 5.0.1. `render` was added on [5.0.0](https://github.com/Shopify/liquid/blob/master/History.md#500--2021-01-06)

**Comment by bluzky** [comment]
Thanks @edgurgel , I have update document for render function and update `liquid` version to 5.0.1 in action configuration file.

**Comment by edgurgel** [comment]
Thanks! Will get a new version released soon 🎉

**Pull request merged**